rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -536,6 +537,49 @@ void getProposedFederationAddress_whenProposedFederationExists_shouldReturnAddre
             assertEquals(expectedAddress, actualProposedFederationAddress.get());
         }
 
+        @ParameterizedTest
+        @EnumSource(FederationMember.KeyType.class)
+        void getProposedFederatorPublicKeyOfType_whenBridgeSupportReturnsEmpty_shouldReturnEmpty(FederationMember.KeyType keyType) {
+            // Arrange
+            var index = 1;
+
+            // Act
+            var actualProposedFederatorPublicKey = bridgeSupport.getProposedFederatorPublicKeyOfType(index, keyType);
+
+            // Assert
+            assertFalse(actualProposedFederatorPublicKey.isPresent());
+        }
+
+        @ParameterizedTest
+        @EnumSource(FederationMember.KeyType.class)
+        void getProposedFederatorPublicKeyOfType_whenProposedFederationExists_shouldReturnAddress(FederationMember.KeyType keyType) {
```

Comment:
An index out of bounds can be thrown, but I don't think this is needed since we would be mocking the exception to be thrown. Also the other type of federations do not test this scenario.

Reply:
The idea would be to test how `BridgeSupport` behaves when `FederationSupport` throws an exception. I think this should be tested it.

Reply:
Alright, I have added a test for this